File: app/src/androidTest/java/com/cottondroid/todoachieved/auth/TestAuthenticationModule.java
package com.cottondroid.todoachieved.auth;

import dagger.Module;
import dagger.Provides;

@Module
public class TestAuthenticationModule {
    private boolean loggedIn;

    public TestAuthenticationModule() {
    }

    public TestAuthenticationModule(boolean loggedIn) {
        this.loggedIn = loggedIn;
    }

    public void setLoggedIn(boolean loggedIn) {
        this.loggedIn = loggedIn;
    }

    @Provides
    public AuthenticationRepository authenticationRepository() {
        return new AuthenticationRepository(null) {
            @Override
            public boolean isLoggedIn() {
                return loggedIn;
            }
        };
    }
}

File: app/src/androidTest/java/com/cottondroid/todoachieved/auth/TestAuthenticationModule.kt
package com.cottondroid.todoachieved.auth

import dagger.Module
import dagger.Provides

@Module
class TestAuthenticationModule(var loggedIn: Boolean = false) {

    @Provides
    fun authenticationRepository(): AuthenticationRepository {
        return object : AuthenticationRepository {
            override val isLoggedIn: Boolean = loggedIn
            override val userId: String? = "userId"
        }
    }
}